### PR TITLE
Implement Event Bridge

### PR DESCRIPTION
**Pull request by deepanjan90**
Fixes: #536

### CODE REVIEW

File: src/main/java/gyro/aws/eventbridge/PlacementStrategy.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    public void setField(String field) {
+        this.field = field;
+    }
+
+    /**
+     * The type of placement strategy.
+     */
+    @Required
```
Comment (harjain99) [inline src/main/java/gyro/aws/eventbridge/PlacementStrategy.java:45]: Add `@ValidStrings`

File: src/main/java/gyro/aws/eventbridge/RedshiftDataParameter.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    public String getDatabase() {
+        return database;
+    }
+
+    public void setDatabase(String database) {
+        this.database = database;
+    }
+
+    /**
+     * The user name for teh database.
```
Comment (harjain99) [inline src/main/java/gyro/aws/eventbridge/RedshiftDataParameter.java:47]:
```suggestion
     * The user name for the database.
```

File: src/main/java/gyro/aws/eventbridge/RedshiftDataParameter.java [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    public String getStatementName() {
+        return statementName;
+    }
+
+    public void setStatementName(String statementName) {
+        this.statementName = statementName;
+    }
+
+    /**
+     * Indicates whether to send an event back to EventBridge after the SQL statement runs.
```
Comment (harjain99) [inline src/main/java/gyro/aws/eventbridge/RedshiftDataParameter.java:96]:
```suggestion
     * When set to ``true``, an event is sent back to EventBridge after the SQL statement runs.
```

File: src/main/java/gyro/aws/eventbridge/TargetResource.java [hunk 4]
```diff
@@ -0,0 +1,395 @@
...
+import gyro.aws.Copyable;
+import gyro.core.resource.Diffable;
+import gyro.core.resource.Updatable;
+import gyro.core.validation.Required;
+import org.apache.commons.lang3.StringUtils;
+import software.amazon.awssdk.services.eventbridge.model.Target;
+
+public class TargetResource extends Diffable implements Copyable<Target> {
+
+    private String arn;
```
Comment (harjain99) [inline src/main/java/gyro/aws/eventbridge/TargetResource.java:28]: Should probably be renamed to something like `sourceArn` for clarity.

**Review by harjain99 — CHANGES_REQUESTED** [review]
Add EventBusRuleResource example file.
Implement TargetFinder
Add `// Read-only` comment for output fields.
Some of the abbreviations are not consistent (e.g. iam vs IAM, ecs vs ECS)

File: src/main/java/gyro/aws/eventbridge/BatchRetryStrategy.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    @Override
+    public void copyFrom(software.amazon.awssdk.services.eventbridge.model.BatchRetryStrategy model) {
+        setAttempts(model.attempts());
+    }
+
+    @Override
+    public String primaryKey() {
+        return String.format("retry attempts - %s", getAttempts());
```
[the remark this answers is not in the document]
Comment (deepanjan90) [reply src/main/java/gyro/aws/eventbridge/BatchRetryStrategy.java:46]: The super parent for this is target, which belongs to Rule as a list, thus the primary key

File: src/main/java/gyro/aws/eventbridge/DeadLetterConfig.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+package gyro.aws.eventbridge;
+
+import gyro.aws.Copyable;
+import gyro.core.resource.Diffable;
+import gyro.core.validation.Required;
+
+public class DeadLetterConfig extends Diffable implements Copyable<software.amazon.awssdk.services.eventbridge.model.DeadLetterConfig> {
+
+    private String arn;
```
[the remark this answers is not in the document]
Comment (deepanjan90) [reply src/main/java/gyro/aws/eventbridge/DeadLetterConfig.java:25]: The @id field for the queue resource is set to the queue url while this diffable needs arn, so is a mismatch

File: src/main/java/gyro/aws/eventbridge/TargetResource.java (hunk 4 above)
Comment (deepanjan90) [reply src/main/java/gyro/aws/eventbridge/TargetResource.java:28]: Keeping it the same being consistent with the api.

**Comment by deepanjan90** [comment]
TargetResource is a subresource thus not needing a finder.

**Review by harjain99 — CHANGES_REQUESTED** [review]

**Review by harjain99 — APPROVED** [review]

**Pull request merged**